src/Microsoft.DotNet.Wpf/src/System.Windows.Primitives/Windows/Win32/Foundation/ComScope.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Windows.Win32.System.Com;
+
+namespace Windows.Win32.Foundation;
+
+/// <summary>
+///  Lifetime management struct for a native COM pointer. Meant to be utilized in a <see langword="using"/> statement
+///  to ensure <see cref="IUnknown.Release"/> is called when going out of scope with the using.
+/// </summary>
+/// <remarks>
+///  <para>
+///   This struct has implicit conversions to T** and void** so it can be passed directly to out methods.
+///   For example:
+///  </para>
+///  <code>
+///   using ComScope&lt;IUnknown&gt; unknown = new(null);
+///   comObject-&gt;QueryInterface(&amp;iid, unknown);
+///  </code>
+///  <para>
+///   Take care to NOT make copies of the struct to avoid accidental over-release.
+///  </para>
+/// </remarks>
+/// <typeparam name="T">
+///  This should be one of the struct COM definitions as generated by CsWin32. Ideally we'd constrain to
+///  <see cref="IUnknown.Interface"/> or some other interface tag to enforce that this is being used around
+///  a struct that is actually a COM wrapper.
+/// </typeparam>
+internal readonly unsafe ref struct ComScope<T> where T : unmanaged, IComIID
```

Comment:
💡 This appears to be using `ref struct` where the intent is closer to `[NonCopyable]`. I recommend changing this to just be `struct` and applying `[NonCopyable]` along with the corresponding analyzer.

Reply:
This is intended to be stack bound so I don't see the value in making this non-ref. There is another pattern for things that aren't that I have not shown yet (`AgileComPointer`).